Week1/Substringtest.java

```diff
@@ -0,0 +1,60 @@
+import static org.testng.Assert.assertEquals;
+
+import org.testng.annotations.Test;
+
+public class Substringtest {
+  LongestSubString s=new LongestSubString();
+  @Test
+  public void testsubString() {
+    String input="aCcBabbcaa";//case sensitive
+   	
+     // assertEquals(actual, expected)
+     assertEquals(4,s.subString("aCcBabbcaa"));
+  } 
+  
+  @Test
+  public void testsubString1() {
+    String input="i am"; //Space 
+   	
+     // assertEquals(actual, expected)
+     assertEquals(0,s.subString("i am"));
+  }
+  
+  @Test
+  public void testsubString2() {
+    String input="a@b#aabb";//special Character 
```

Comment:
In each of these methods, you create an input variable, but do not use it, instead copying the string again in the assert. You should not have variables that are never used in your code!